src/validate-with-options.js

```diff
@@ -0,0 +1,63 @@
+import validateJs from 'validate.js'
+import {
+  formatErrors,
+  flatToNested,
+} from './utils'
+
+/**
+ * A wrapper around the `validate` function exported from
+ * {@link https://validatejs.org/|Validate JS} to make it work seamlessly with
+ * {@link http://redux-form.com/|Redux Form}.
+ * 
+ * @param {Object} constraints - A 'flat' object containing constraints in the
+ * format specified by Validate JS. These are key-value pairs where the keys
+ * correspond to keys in the data that will be validated. This is a 'flat'
+ * object in that nested data must be accessed using a string path
+ * (ex. 'foo.bar') as the key.
+ * @param {Object} options - An object to pass in any options specified by `validateJS`.
+ * @param {Object} values - A nested object containing values to be validated.
+ * 
+ * @returns {Object} errors - A nested object of errors that will be passed to redux form.
+ * 
+ * @example
+ * 
+ * const values = {
+ *   name: 'Foo',
+ *   address: {
+ *     zip: '12'
+ *   }
+ * }
+ * 
+ * const options = {
+ *   fullMessages: true,
+ * }
+ * 
+ * const constraints = {
+ *   name: {
+ *     presence: true
+ *   },
+ *   'address.zip': {
+ *     presence: true,
+ *     length: { is: 5 }
+ *   }
+ * }
+ * 
+ * validate(constraints, options, values)
+ * 
+ * // {
+ * //   address: {
+ * //     zip: ['Zip is the wrong length (should be 5 characters)']
+ * //   }
+ * // }
+ */
+
+function validateWithOptions (constraints, values, options={}) {
```

Comment:
Let's just have validateWithOptions internally call `validate` so we don't duplicate logic.

Reply:
Sorry if I'm totally missing something here -- how will I pass through the options this way?

Reply:
Oh yeah, didn't think about that haha.
You could export the internal validate function as a named export and use that.

Reply:
Wait, sorry, that doesn't make sense.
You can make `validate` call `validateWithOptions` though.